File: base/000/Solver.java
import java.util.Scanner;

public class Solver{
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		double reg = 0.0;

		while(true){
			String line = scanner.nextLine();
			System.out.println("$" + line);
			String ui[] = line.split(" "); //UI == user input
			String cmd = ui[0];

			if(line.equals("end")){
				break;
			}else if(cmd.equals("show")){
				System.out.printf("%.2f\n", reg);
			}else if(cmd.equals("add")){
				reg += Double.parseDouble(ui[1]);
			}else if(cmd.equals("mult")){
				reg *= Double.parseDouble(ui[1]);
			}else if(cmd.equals("div")){
				double value = Double.parseDouble(ui[1]);
				if(value == 0.0)
					System.out.println("fail: division by zero");
				else
					reg /= value;
			}else if(cmd.equals("addm")) {
				for(int i = 1; i < ui.length; i++){
					reg += Double.parseDouble(ui[i]);
				}
			}else{
				System.out.println("fail: command not found");
			}
		}
		scanner.close();
	}
}

File: trash/000/solver.cpp
#include <iostream>
#include <sstream>
#include <cstdio>
using namespace std;

int main(){
    string line;
    string cmd;
    float acc = 0;
    while(true){
        getline(cin, line);
        cout << "$" << line << "\n";
        stringstream ss(line);
        ss >> cmd;

        if(line == "end"){
            break;
        }else if(cmd == "show"){
            printf("%.2f\n", acc);
        }else if(cmd == "add"){
            float value;
            ss >> value;
            acc += value;
        }else if(cmd == "mult"){
            float value;
            ss >> value;
            acc *= value;
        }else if(cmd == "div"){
            float value;
            ss >> value;
            if(value == 0)
                puts("fail: division by zero");
            else
                acc /= value;
        }else if(cmd == "addm"){
            float value;
            while(ss >> value){
                acc += value;
            }
        }else{
            puts("fail: command not found");
        }

    }

}